engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/EndEventProcessor.java

```diff
@@ -313,4 +314,29 @@ public void onComplete(final ExecutableEndEvent element, final BpmnElementContex
               failure -> incidentBehavior.createIncident(failure, completing));
     }
   }
+
+  private final class CompensationBehaviour implements EndEventBehavior {
+
+    @Override
+    public boolean isSuitableForEvent(final ExecutableEndEvent element) {
+      return element.isCompensationEvent();
+    }
+
+    @Override
+    public void onActivate(final ExecutableEndEvent element, final BpmnElementContext activating) {
+      final var activated =
+          stateTransitionBehavior.transitionToActivated(activating, element.getEventType());
+      final var completing = stateTransitionBehavior.transitionToCompleting(activated);
+      onComplete(element, completing);
+    }
+
+    @Override
+    public void onComplete(final ExecutableEndEvent element, final BpmnElementContext completing) {
+      stateTransitionBehavior
+          .transitionToCompleted(element, completing)
+          .ifRightOrLeft(
+              stateTransitionBehavior::completeElement,
+              failure -> incidentBehavior.createIncident(failure, completing));
+    }
```

Comment:
:x: We're making an invalid transition. The end event is in the state `COMPLETED` and we write a `COMPLETE` command. 

In the record stream, you can see that the `COMPLETE` command is rejected. 

```
 - #26->#20 K10 - SEQUENCE_FLOW "sequenc..1b104eb" in <process "compens..process"[K03]>
C PI        ACTIVATE       - #27->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
E PI        ACTIVATING     - #28->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
E PI        ACTIVATED      - #29->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
E PI        COMPLETING     - #30->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
E PI        COMPLETED      - #31->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
C PI        COMPLETE       - #32->#20 K03 - PROCESS "compens..process" in <process "compens..process"[K03]>
C PI        COMPLETE       - #33->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]>
E PI        COMPLETING     - #34->#20 K03 - PROCESS "compens..process" in <process "compens..process"[K03]>
E PI        COMPLETED      - #35->#20 K03 - PROCESS "compens..process" in <process "compens..process"[K03]>
R PI        COMPLETE       - #36->#20 K11 - END_EVENT "compens..n-event" in <process "compens..process"[K03]> !INVALID_STATE (Expected element instance with key '2251799813685259' to be present in state but not found.)
```

I recommend that we follow the other behaviors and call `stateTransitionBehavior.takeOutgoingSequenceFlows(element, completed)`.

Reply:
Thanks for spotting 👀